game/static/game/js/level_editor.js

```diff
@@ -2972,14 +2962,15 @@ ocargo.LevelEditor = function(levelId) {
         this.image = drawing.createCowImage(data.group.type);
         this.valid = false;
 
-
         if ( data.coordinates && data.coordinates.length > 0 ) {
-            var coordinates = new ocargo.Coordinate(data.coordinates[0].x, data.coordinates[0].y);
-            this.controlledNode = ocargo.Node.findNodeByCoordinate(coordinates, nodes);
-
-            if (this.controlledNode) {
-                this.valid = true;
-                drawing.setCowImagePosition(coordinates, this.image, this.controlledNode);
+            this.coordinate = new ocargo.Coordinate(data.coordinates[0].x, data.coordinates[0].y);
+            this.valid = isValidDraggedCowPlacement(this.coordinate, this);
+            
+            if (this.isOnRoad()) {
+                const controlledNode = ocargo.Node.findNodeByCoordinate(coordinates, nodes);
+                drawing.setCowImagePosition(this.coordinate, this.image, controlledNode);
+            } else {
+                this.image.transform('...t' + (-paper.scrollLeft())  +  ',' +  paper.scrollTop());
```

Comment:
@faucomte97 to test these code blocks I need to sign in to the website in my local setup. Can you tell me how to sign up? 

I have manually tested other blocks locally. What else do you suggest me to do to make sure I am not breaking any existing functionalities?

Reply:
Hi @RaihanRasheedApurbo, you can either:
- log in using any of the test teacher accounts already existing in the DB (details of those test accounts can be found here: https://github.com/ocadotechnology/codeforlife-portal/blob/master/portal/migrations/0001_squashed_0041_new_news.py)
- create a user of your own by following the registration process. The content of the emails you'd normally get during the registration process will get outputted to your terminal which is where you'll find the verification link, for example.

Let me know if that makes sense, option 1 is usually the easier one.

Then, regarding your other question - our automated test pipeline here on GitHub will test the functionality of the website, so if anything outside of your test broke, the pipeline will fail. I will approve the pipeline now